emily/lambda/src/main.rs

```diff
@@ -0,0 +1,175 @@
+use aws_lambda_events::apigw::{ApiGatewayProxyRequest, ApiGatewayProxyResponse};
+use aws_lambda_events::encodings::Body;
+use emily::models::{CreateDepositRequestContent, DepositParameters};
+use http::{header, HeaderValue};
+use lambda_runtime::{service_fn, Error, LambdaEvent};
+use http::{HeaderMap, Method};
+
+/// Asynchronously handles incoming API Gateway events and dispatches them
+/// to the correct method based on the route and method.
+///
+/// # Arguments
+/// * `event` - A LambdaEvent containing the ApiGatewayProxyRequest.
+///
+/// # Returns
+/// A result containing the API Gateway Proxy Response or an error.
+async fn handle_event(
+    event: LambdaEvent<ApiGatewayProxyRequest>
+) -> Result<ApiGatewayProxyResponse, Error> {
+
+    // Extract base data.
+    let resource = event.payload.resource.unwrap_or_default();
+    let http_method = event.payload.http_method;
+    let body = event.payload.body.clone();
+
+    // Dispatch based on API call.
+    match (resource.as_str(), http_method) {
+        ("/deposits", Method::POST) => {
+
+            // Deserialize the create deposit request.
+            let request: CreateDepositRequestContent = body
+                .map(|serailzed_request| {
+                    serde_json::from_str::<CreateDepositRequestContent>(serailzed_request.as_str())
+                    .unwrap() // TODO: Handle errors.
+                })
+                .unwrap(); // TODO: Handle errors.
+
+            // Generate dummy response with all the necesary fields, taking
+            // the identifiers from the request.
+            let response = emily::models::CreateDepositResponseContent {
+                bitcoin_txid: request.bitcoin_txid.to_string(),
+                bitcoin_tx_output_index: request.bitcoin_tx_output_index,
+                recipient: "MOCK_RECIPIENT".to_string(),
+                amount: 11111.0,
+                status: emily::models::OpStatus::Pending,
+                status_message: "MOCK_CREATE_DEPOSIT_RESPONSE".to_string(),
+                parameters: Box::new(DepositParameters {
+                    lock_time: Some(22222.0),
```

Comment:
Wow, that's a long time.

Reply:
Wait why is `amount`, `lock_time`, and `max_fee` not an integers?

Reply:
It's just what the autogenerator provided, but I am not sure why it does that, the template specifically says `Integer`.

Reply:
I don't know how, but I bet this has to do with JavaScript using the same "number" type for anything numeric :sweat_smile:.